code/go/ecs/x509.go

```diff
@@ -0,0 +1,114 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+// Code generated by scripts/gocodegen.go - DO NOT EDIT.
+
+package ecs
+
+import (
+	"time"
+)
+
+// This implements the common core fields for x509 certificates. This
+// information is likely logged with TLS sessions, digital signatures found in
+// executable binaries, S/MIME information in email bodies, or analysis of
+// files on disk. When only a single certificate is logged in an event, it
+// should be nested under `file`. When hashes of the DER-encoded certificate
+// are available, the `hash` data set should be populated as well (e.g.
+// `file.hash.sha256`). For events that contain certificate information for
+// both sides of the connection, the x509 object could be nested under the
+// respective side of the connection information (e.g. `tls.server.x509`).
+type X509 struct {
+	// Version of x509 format.
+	VersionNumber string `ecs:"version_number"`
+
+	// Unique serial number issued by the certificate authority. For
+	// consistency, if this value is alphanumeric, it should be formatted
+	// without colons and uppercase characters.
+	SerialNumber string `ecs:"serial_number"`
```

Comment:
Per https://tools.ietf.org/html/rfc5280#section-4.1.2.2 this should always be an integer. String is still appropriate since it may be up to 20 octets, but I think we should specify that we expect a base10 string representation.